contributors/design-proposals/propagation.md

```diff
@@ -142,13 +153,41 @@ distros.
 1. (From @euank) Changing those mountflags may make docker even less stable,
 this may lock up kernel accidentally or potentially leak mounts.
 
+1. (From @jsafrane) Typical container that needs to mount something needs to
+see host's `/dev` and `/sys` as HostPath volumes. This would make them shared
+without any way to opt-out. Docker creates a new `/dev/shm` in the
+container, which gets propagated to the host, shadowing host's `/dev/shm`.
+Similarly, systemd running in a container is very picky about `/sys/fs/cgroup`
+and something prevents it from starting if `/sys` is shared.
 
 ## Decision
 
-We will take 'Make HostPath shared for privileged containers, slave for
-non-privileged', an environment check and an WARNING log will be emitted about
-whether propagation mode is supported.
-
+* We will take 'Add an option in VolumeMount API' (with an annotation during
```

Comment:
There are several fields [1] that apply to multiple VolumeSources, and the fields are replicated amongst the applicable sources. On the other hand, all the fields in VolumeMount are universally applicable to all volumes. I think it would be more consistent in this case to just include the propagation mode in both the HostPath and EmptyDir VolumeSources.

[1] FSType, ReadOnly, DefaultMode